src/webapi/date.rs

```diff
@@ -1,22 +1,903 @@
+    #[inline]
+    fn to_locale_date_string( &self, _locales: Option< &str >, _options: Option< &HashMap< String, String > >) -> String {
+        match _locales {
+            Some(_locales) => match _options {
+                Some(_options) => return js! (
+                    var _options = @{_options};
+
+                    return @{self.as_ref()}.toLocaleDateString(@{_locales}, _options);
+                 ).try_into().unwrap(),
+
+                 None => return js! (
+                     return @{self.as_ref()}.toLocaleDateString(@{_locales});
+                 ).try_into().unwrap()
+            },
+
+            None => return js! ( return @{self.as_ref()}.toLocaleDateString(); ).try_into().unwrap()
+        }
+
+    }
+
+    /// NON-STANDARD
+    /// fn to_locale_format( &self ) -> f64 { }
+
+    /// to_locale_string
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toLocaleString)
+    #[inline]
+    fn to_locale_string( &self, _locales: Option< &str >, _options: Option< &HashMap< String, String > >) -> String {
+        match _locales {
+            Some(_locales) => match _options {
+                Some(_options) => return js! (
+                    var _options = @{_options};
+
+                    return @{self.as_ref()}.toLocaleString(@{_locales}, _options);
+                ).try_into().unwrap(),
+
+                None => return js! (
+                    return @{self.as_ref()}.toLocaleString(@{_locales});
+                ).try_into().unwrap()
+            },
+
+            None => return js! ( return @{self.as_ref()}.toLocaleString(); ).try_into().unwrap()
+        }
+    }
+
+    /// to_locale_time_string
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toLocaleTimeString)
+    #[inline]
+    fn to_locale_time_string( &self, _locales: Option< &str >, _options: Option< &HashMap< String, String > > ) -> String {
+        match _locales {
+            Some(_locales) => match _options {
+                Some(_options) => return js! (
+                    var _options = @{_options};
+
+                    return @{self.as_ref()}.toLocaleTimeString(@{_locales}, _options);
+                ).try_into().unwrap(),
+
+                None => return js! (
+                    return @{self.as_ref()}.toLocaleTimeString(@{_locales});
+                ).try_into().unwrap()
+            },
+
+            None => return js! ( return @{self.as_ref()}.toLocaleTimeString(); ).try_into().unwrap()
+        }
+    }
+
+    /// NON-STANDARD
+    /// fn to_source( &self ) -> f64 { }
+
+    ///
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toString)
+    #[inline]
+    fn toString( &self ) -> String {
```

Comment:
It would probably be more Rusty to do an `impl From< &Date > for String` or something rather than this.